account-kit/logging/src/index.ts

```diff
@@ -1,15 +1,54 @@
 import { createClientLogger } from "./client.js";
 import { createServerLogger } from "./server.js";
-import type { EventLogger, EventsSchema } from "./types";
+import type { EventLogger, EventsSchema, LoggerContext } from "./types";
 
 export type * from "./types.js";
 
-export function createLogger<
-  Schema extends EventsSchema
->(): EventLogger<Schema> {
-  if (typeof window === "undefined") {
-    return createServerLogger();
-  }
+export function createLogger<Schema extends EventsSchema = []>(
+  context: LoggerContext
+): EventLogger<Schema>;
 
-  return createClientLogger();
+export function createLogger(context: LoggerContext): EventLogger {
+  const innerLogger =
+    typeof window === "undefined"
+      ? createServerLogger(context)
+      : createClientLogger(context);
+
+  const logger: EventLogger = {
+    ...innerLogger,
+    profiled<TFunc extends (...args: any[]) => Promise<unknown> | unknown>(
+      name: string,
+      func: TFunc
+    ): TFunc {
+      return new Proxy(func, {
+        apply(target, thisArg, argArray) {
+          const start = Date.now();
+          const result = target.apply(thisArg, argArray);
+          if (result instanceof Promise) {
+            return result.then((res) => {
```

Comment:
Are you interested in logging in the case where the promise rejects?

Reply:
Yea I was thinking about that and it's in the spec as an open question. I'm wondering if error timings are helpful at all or if that just ends up adding noise?

Reply:
Yeah, maybe it would just make the data confusing if it were mixed in with the other timings.